### PR TITLE
DS-121 Improve search bar close icon

### PR DESCRIPTION
**Pull request by MrVorgoth**
Jira: https://livechatinc.atlassian.net/browse/DS-121

Hey, according to the above task the close icon in search bar is disappearing quicker from the DOM than some helper function/hook a developer wants to use and it is causing problems. I've prepared a fix that icon is always visible in the DOM (`display: none/block`) and even if user will turn this icon on through dev console then nothing will happen (blocked handlers on this icon).

Moreover I'm trying new concept of `data-element` attributes to help user find elements inside given component. WDYT about this approach?

PS: In the code there are some prettier code formatting changes

### CODE REVIEW

File: packages/design-system/src/components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -268,14 +280,15 @@ SearchBar.propTypes = {
...
+  onBlur: PropTypes.func,
+  forwardedRef: PropTypes.func
 };
 
 SearchBar.defaultProps = {
   placeholder: 'Search...',
   debounceTime: 0
 };
 
-export default React.forwardRef(function SearchBarWithRef(props, ref) {
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/SearchBar/SearchBar.js:279]: if you just use an anonymous arrow function here the component will be harder to find in React Dev Tools (https://reactjs.org/docs/forwarding-refs.html#displaying-a-custom-name-in-devtools) + possible error message will be less helpful

File: packages/design-system/src/components/SearchBar/SearchBar.js [hunk 2]
```diff
@@ -268,14 +280,15 @@ SearchBar.propTypes = {
   onCollapse: PropTypes.func,
   onExpand: PropTypes.func,
   onFocus: PropTypes.func,
-  onBlur: PropTypes.func
+  onBlur: PropTypes.func,
+  forwardedRef: PropTypes.func
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/SearchBar/SearchBar.js:284]: What about `React.RefObject`?

File: packages/design-system/src/components/SearchBar/style.scss [hunk 3]
```diff
@@ -50,6 +51,10 @@ $base-class: "search-bar";
       box-sizing: border-box;
       border: 1px solid #8db4ee;
     }
+
+    &--disable {
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/SearchBar/style.scss:55]:
```suggestion
    &--hidden {
```
`disable` usually means something different

File: packages/design-system/src/components/SearchBar/SearchBar.js [hunk 4]
```diff
@@ -212,20 +225,19 @@ export class SearchBar extends React.PureComponent {
...
+          <CloseIcon
+            width="18px"
+            height="18px"
+            onClick={shouldDisplayClearButton ? this.handleClear : null}
+            onKeyDown={
+              shouldDisplayClearButton ? this.handleClearButtonKeyDown : null
+            }
+            className={closeIconClassName}
+            tabIndex="0"
+            data-element="close-icon"
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/SearchBar/SearchBar.js:239]: I was thinking about that recently and read a couple of articles about e2e tests and selectors. 
`className` is guaranteed to not change in design-system lib but maybe we should think about this property just as a way to style things. `data-testId` could inform about what is this element and it could be used in tests. Also there is a smaller chance that it will change (than class name). It will be the same until the element is removed or replaced with something else.

To sum it up - let's add `data-` attribute but it would be better to use the convention proposed in AA + use `button` word instead of `icon` since it's clickable:
```suggestion
            data-testId="close-button"
```

**Review by kamilmateusiak — CHANGES_REQUESTED** [review]
some minor things to be fixed

**Review by quarties — APPROVED** [review]
LGTM (except @kamilmateusiak comments) 💪

File: packages/design-system/src/components/SearchBar/SearchBar.js (hunk 4 above)
Comment (MrVorgoth) [reply packages/design-system/src/components/SearchBar/SearchBar.js:239]: I'm all in for that approach :)

**Review by kamilmateusiak — APPROVED** [review]
